File: archive/web-react-vite/src/types/book.ts
export type BookStatus = "wishlist" | "reading" | "finished" | "paused" | "dropped";

export interface Book {
  id: number;
  title: string;
  author?: string | null;
  cover_url?: string | null;
  category?: string | null;
  status: BookStatus;
  rating?: number | null;
  tags: string[];
  short_comment?: string | null;
  review?: string | null;
  start_date?: string | null;
  finish_date?: string | null;
  created_at: string;
  updated_at: string;
  is_deleted: boolean;
  deleted_at?: string | null;
}

export interface BookPayload {
  title: string;
  author?: string | null;
  cover_url?: string | null;
  category?: string | null;
  status: BookStatus;
  rating?: number | null;
  tags: string[];
  short_comment?: string | null;
  review?: string | null;
  start_date?: string | null;
  finish_date?: string | null;
}

export interface BookFilters {
  status?: BookStatus;
  keyword?: string;
  tag?: string;
}

export const BOOK_STATUS_LABELS: Record<BookStatus, string> = {
  wishlist: "想读",
  reading: "在读",
  finished: "已读",
  paused: "暂停",
  dropped: "弃读",
};

export const BOOK_STATUSES: BookStatus[] = [
  "wishlist",
  "reading",
  "finished",
  "paused",
  "dropped",
];

File: app/src/main/java/com/example/readtrace/model/Book.kt
package com.example.readtrace.model

data class Book(
    val id: Long = 0,
    val title: String,
    val author: String? = null,
    val coverUrl: String? = null,
    val category: String? = null,
    val status: BookStatus = BookStatus.WISHLIST,
    val mediaType: MediaType = MediaType.BOOK,
    val rating: Double? = null,
    val tags: List<String> = emptyList(),
    val shortComment: String? = null,
    val review: String? = null,
    val startDate: String? = null,
    val finishDate: String? = null,
    val buyChannel: String? = null,
    val shelfLocation: String? = null,
    val bindingType: String? = null,
    val buyPrice: Double? = null,
    val createdAt: String = "",
    val updatedAt: String = "",
    val isDeleted: Boolean = false,
    val deletedAt: String? = null,
    // 外部导入来源（v4.2.14）：'bangumi' 等，NULL 视为手动录入；source_id 为来源条目唯一 ID
    val sourceType: String? = null,
    val sourceId: String? = null,
    // 远程评分（如 Bangumi 全站评分 0~10），与个人评分 rating 分离互不覆盖
    val remoteRating: Double? = null,
    // 简介正文（外部导入填充，用户可自行修改）
    val description: String? = null,
)
